sklego/neighbors.py

```diff
@@ -0,0 +1,122 @@
+from multiprocessing import cpu_count
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.externals.joblib import Parallel, delayed
+from sklearn.neighbors import KernelDensity
+from sklearn.utils import check_X_y
+from sklearn.utils.multiclass import unique_labels
+from sklearn.utils.validation import check_is_fitted, check_array, FLOAT_DTYPES
+
+
+class BayesianKernelDensityClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(
+        self,
+        bandwidth=0.2,
+        kernel="gaussian",
+        algorithm="auto",
+        metric="euclidean",
+        atol=0,
+        rtol=0,
+        breath_first=True,
+        leaf_size=40,
+        metric_params=None,
+        n_jobs=None,
+    ):
+        """
+        Bayesian Classifier that uses Kernel Density Estimations to generate the joint distribution
+
+        All parameters of the model are an exact copy of the parameters in scikit-learn.
+        """
+        self.bandwidth = bandwidth
+        self.kernel = kernel
+        self.algorithm = algorithm
+        self.metric = metric
+        self.atol = atol
+        self.rtol = rtol
+        self.breath_first = breath_first
+        self.leaf_size = leaf_size
+        self.metric_params = metric_params
+
+        if n_jobs is not None:
```

Comment:
@MBrouns I think this needs to be tested in `.fit()` no?

Reply:
Yep!

Reply:
Also, I just checked some other sklearn estimators that have n_jobs, and it seems like`None` actually means only 1 job, -1 means all possible cores

Reply:
Thanks for doing all this research for me give me a sec to make those changes

Reply:
I just found that (at least for RandomForests) no checks are done on the `n_jobs` argument at all. It seems that all negative ints use the cores

The only thing that appears to break the code is floats. Since using a float raises a really obscure error (I assume with how job lists get sliced?) I think floats should be the only thing I check for.

Ugly error with floats looks like this.
```python
Traceback (most recent call last):
  File "C:\Users\antho\Miniconda3\lib\site-packages\joblib\parallel.py", line 797, in dispatch_one_batch
    tasks = self._ready_batches.get(block=False)
  File "C:\Users\antho\Miniconda3\lib\queue.py", line 167, in get
    raise Empty
_queue.Empty

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/antho/.PyCharm2019.3/config/scratches/scratch_2.py", line 25, in <module>
    RandomForestClassifier(5000, n_jobs=1.23).fit(x, y)
  File "C:\Users\antho\Miniconda3\lib\site-packages\sklearn\ensemble\_forest.py", line 383, in fit
    for i, t in enumerate(trees))
  File "C:\Users\antho\Miniconda3\lib\site-packages\joblib\parallel.py", line 1004, in __call__
    if self.dispatch_one_batch(iterator):
  File "C:\Users\antho\Miniconda3\lib\site-packages\joblib\parallel.py", line 808, in dispatch_one_batch
    islice = list(itertools.islice(iterator, big_batch_size))
ValueError: Stop argument for islice() must be None or an integer: 0 <= x <= sys.maxsize.
```

Reply:
I'm actually ok with ignoring floats. We can expect some level of competency for our users I would say. @koaning?